connector/src/main/java/io/pixelsdb/pixels/trino/vector/lshnns/CachedLSHIndex.java

```diff
@@ -0,0 +1,259 @@
+package io.pixelsdb.pixels.trino.vector.lshnns;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import io.pixelsdb.pixels.trino.PixelsColumnHandle;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+public class CachedLSHIndex {
+
```

Comment:
We should use the Storage API to access the underlying storage system such as S3, instead of directly using the S3 client.